### PR TITLE
add pipeline name validation

### PR DESCRIPTION
**Pull request by DSmolonogov**
Fixes https://github.com/ozontech/file.d/issues/179

### CODE REVIEW

File: fd/file.d.go [hunk 1]
```diff
@@ -230,20 +230,18 @@ func (f *FileD) getStaticInfo(pipelineConfig *cfg.PipelineConfig, pluginKind pip
 		return nil, fmt.Errorf("no %s plugin provided", pluginKind)
 	}
 	t := configJSON.Get("type").MustString()
+	configJSON.Del("type")
```
Comment (vano144) [inline fd/file.d.go:233]: need comment here

File: cfg/config_test.go [hunk 2]
```diff
@@ -479,3 +479,17 @@ func TestParseDefaultInt(t *testing.T) {
 		assert.Equal(t, tc.expected, tc.s.T, "wrong value tc: %d", i)
 	}
 }
+
+func TestPipelineValidatorValid(t *testing.T) {
+	testName := []string{"pipeline_name", "PipeLine_NAME", "pipelinename", "PIPELINENAME"}
+	for _, tl := range testName {
+		assert.NoErrorf(t, validatePipelineName(tl), `pipeline name "%s" should be valid`, tl)
```
Comment (ansakharov) [inline cfg/config_test.go:486]: Maybe replace: assert.NoErrorf with assert.NoError(t, validatePipelineName(tl))

File: cmd/file.d_test.go [hunk 3]
```diff
@@ -212,3 +216,60 @@ func TestThatPluginsAreImported(t *testing.T) {
...
 	}
 }
+
+type testConfig struct {
+	factory    func() (pipeline.AnyPlugin, pipeline.AnyConfig)
+	configJSON string
+}
+
+func TestConfigParseValid(t *testing.T) {
+	testList := []testConfig{
```
Comment (ansakharov) [inline cmd/file.d_test.go:226]: You can add t.Parallel() here

File: cmd/file.d_test.go [hunk 4]
```diff
@@ -212,3 +216,60 @@ func TestThatPluginsAreImported(t *testing.T) {
...
+			factory:    gelf.Factory,
+			configJSON: `{"endpoint":"graylog.svc.cluster.local:12201","reconnect_interval":"1m","default_short_message_value":"message isn't provided"}`,
+		},
+		{
+			factory:    splunk.Factory,
+			configJSON: `{"endpoint":"splunk_endpoint","token":"value_token"}`,
+		},
+	}
+	for _, tl := range testList {
+		_, config := tl.factory()
```
Comment (ansakharov) [inline cmd/file.d_test.go:249]: add `name` property to testList - it represents what case you testing.
And do:
```
t.Run(tl.name, func(t *testing.T) {
  _, config := tl.factory()
 err := fd.DecodeConfig(config, []byte(tl.configJSON))
  assert.NoError(t, err, "shouldn't be an error")
}
```

File: cmd/file.d_test.go [hunk 5]
```diff
@@ -212,3 +216,60 @@ func TestThatPluginsAreImported(t *testing.T) {
...
+		err := fd.DecodeConfig(config, []byte(tl.configJSON))
+		assert.NoError(t, err, "shouldn't be an error")
+	}
+}
+
+func TestConfigParseInvalid(t *testing.T) {
+	testList := []testConfig{
```
Comment (ansakharov) [inline cmd/file.d_test.go:256]: same as above - add test name, t.Run, t.Parallel()

**Review by vadimalekseev — APPROVED** [review]

**Review by ansakharov — APPROVED** [review]
LGTM

**Pull request merged**